metrics/stackdriver_exporter.go

```diff
@@ -47,13 +58,30 @@ var (
 	// In unit tests this is set to a fake one to avoid calling actual Google API
 	// service.
 	newStackdriverExporterFunc func(stackdriver.Options) (view.Exporter, error)
+
+	// getStackdriverSecretFunc is the function used to create read a Kubernetes
+	// secret for Stackdriver credentials into memory.
+	// In product usage, this is always set to function sdconfig.getStackdriverSecret.
+	// In unit tests this is set to a fake one because an in-cluster kubeclient is required.
+	getStackdriverSecretFunc func(config *stackdriverClientConfig) (*corev1.Secret, error)
+
+	// kubeclient is the in-cluster Kubernetes kubeclient, which is lazy-initialized on first use.
+	kubeclient *kubernetes.Clientset
+	// initClientOnce is the lazy initializer for kubeclient.
+	initClientOnce sync.Once
+	// kubeclientInitErr capture an error during initClientOnce
+	kubeclientInitErr error
```

Comment:
I didn't find this is read anywhere after being set. Is it necessary to keep it in memory?

Reply:
If initializing the kubeclient fails, getStackdriverSecret will end up returning this error on every subsequent call, which is logged as a warning whenever stackdriver exporter is updated.

Reply:
Can't we just make `kubeclientInitErr` a local var in `ensureKubeclient`?

Reply:
Because `initClientOnce` will only run once, after the first error `initClientOnce` will be a no-op. 

If I make the `kubeclientInitErr` local to `ensureKubeclient`, `kubeclientInitErr` will be reset on subsequent calls to `ensureKubeclient` and start to return nil  (or whatever the local initialization value is), even though kubeclient is still not initialized.

Reply:
Got it.